source/MaterialXFormat/Util.cpp

```diff
@@ -139,7 +139,7 @@ StringSet loadLibraries(const FilePathVec& libraryFolders,
     return loadedLibraries;
 }
 
-void flattenFilenames(DocumentPtr doc, const FileSearchPath& searchPath, StringResolverPtr customResolver)
+void flattenFilenames(DocumentPtr doc, const FileSearchPath& searchPath, StringResolverPtr customResolver, const FilePathPredicate& skipFlattening)
```

Comment:
Two thoughts on this new argument to `flattenFilenames`:
- I don't yet see any usage of this argument within the MaterialX codebase, so I would double-check that it's truly needed.
- If the change is still needed, I would recommend `pathPredicate` as a clearer argument name.

Reply:
If you end up naming the argument `pathPredicate`, then it would also make sense to skip paths for which the predicate returns `false`, rather than skipping paths for which it returns `true`.  That would follow the usage of other predicate functions in the MaterialX codebase.